packages/block-editor/src/components/global-styles/hooks.js

```diff
@@ -204,63 +204,128 @@ export function useGlobalStyle(
+					select( blocksStore ).getBlockType( blockName )?.supports,
 			};
 		},
-		[ name, element ]
+		[ blockName, element ]
 	);
 
-	return supportedPanels;
-}
+	return useMemo( () => {
+		const updatedSettings = { ...parentSettings };
 
-/**
- * Given a settings object and a list of supported panels,
- * returns a new settings object with the unsupported panels removed.
- *
- * @param {Object}   settings Settings object.
- * @param {string[]} supports Supported style panels.
- *
- * @return {Object} Merge of settings and supports.
- */
-export function overrideSettingsWithSupports( settings, supports ) {
-	const updatedSettings = { ...settings };
-
-	if ( ! supports.includes( 'fontSize' ) ) {
-		updatedSettings.typography = {
-			...updatedSettings.typography,
-			fontSizes: {},
-			customFontSize: false,
-		};
-	}
-
-	if ( ! supports.includes( 'fontFamily' ) ) {
-		updatedSettings.typography = {
-			...updatedSettings.typography,
-			fontFamilies: {},
-		};
-	}
+		if ( ! supportedStyles.includes( 'fontSize' ) ) {
+			updatedSettings.typography = {
+				...updatedSettings.typography,
+				fontSizes: {},
+				customFontSize: false,
+			};
+		}
 
-	[
-		'lineHeight',
-		'fontStyle',
-		'fontWeight',
-		'letterSpacing',
-		'textTransform',
-		'textDecoration',
-	].forEach( ( key ) => {
-		if ( ! supports.includes( key ) ) {
+		if ( ! supportedStyles.includes( 'fontFamily' ) ) {
 			updatedSettings.typography = {
 				...updatedSettings.typography,
-				[ key ]: false,
+				fontFamilies: {},
 			};
 		}
-	} );
 
-	return updatedSettings;
+		[
+			'lineHeight',
+			'fontStyle',
+			'fontWeight',
+			'letterSpacing',
+			'textTransform',
+			'textDecoration',
+		].forEach( ( key ) => {
+			if ( ! supportedStyles.includes( key ) ) {
+				updatedSettings.typography = {
+					...updatedSettings.typography,
+					[ key ]: false,
+				};
+			}
+		} );
+
+		[ 'contentSize', 'wideSize' ].forEach( ( key ) => {
+			if ( ! supportedStyles.includes( key ) ) {
+				updatedSettings.layout = {
+					...updatedSettings.layout,
+					[ key ]: false,
+				};
+			}
+		} );
+
+		[ 'padding', 'margin', 'blockGap' ].forEach( ( key ) => {
+			if ( ! supportedStyles.includes( key ) ) {
+				updatedSettings.spacing = {
+					...updatedSettings.spacing,
+					[ key ]: false,
+				};
+			}
+		} );
+
+		if ( ! supportedStyles.includes( 'padding' ) ) {
+			updatedSettings.spacing = {
+				...updatedSettings.spacing,
+				padding: false,
+			};
+		}
+
+		if ( ! supportedStyles.includes( 'margin' ) ) {
+			updatedSettings.spacing = {
+				...updatedSettings.spacing,
+				margin: false,
+			};
+		}
+
+		if ( ! supportedStyles.includes( 'blockGap' ) ) {
+			updatedSettings.spacing = {
+				...updatedSettings.spacing,
+				blockGap: false,
+			};
+		}
+
+		if ( ! supportedStyles.includes( 'minHeight' ) ) {
+			updatedSettings.dimensions = {
+				...updatedSettings.dimensions,
+				minHeight: false,
+			};
+		}
+
+		[ 'padding', 'margin', 'blockGap' ].forEach( ( key ) => {
```

Comment:
Should this be consolidated with the previous loop through `padding`, `margin`, and `blockGap`?